src/main/webapp/app/exercises/programming/shared/service/external-cloning.service.ts

```diff
@@ -12,26 +13,21 @@ export class ExternalCloningService {
     }
 
     /**
-     * Builds an url to clone a Repository in IntelliJ
-     * Structure: jetbrains://idea/checkout/git?idea.required.plugins.id=Git4Idea&checkout.repo=:RepoUrl
+     * Builds an url to clone a Repository in the corresponding ide
      * @param cloneUrl the url of the repository to clone
+     * @param ide the ide with the deeplink to build the url for
      */
-    buildJetbrainsUrl(cloneUrl: string | undefined): string | undefined {
+    // TODO shift exercise Programming Language Logic here
+    buildIDEUrl(cloneUrl: string | undefined, ide: Ide): string | undefined {
         if (!cloneUrl) {
+            console.log('cloneUrl is undefined');
```

Comment:
Is that a forgotten debug log?
If not, we either use the alert service for error messages or use sentry (`captureException`)

Reply:
the alert service is to communicate the error to the user as a pop up right?
I just want to log it similar to a server log

Reply:
In that case you need the `captureException`, since the browser console is not logged

Reply:
as this case should never happen anyways, @JohannesWt @pzdr7 suggested just removing the log